docs/technical_reference/tools/parallel_manager.rst

```diff
@@ -0,0 +1,105 @@
+
+Usage
+--------
+
+End User
+~~~~~~~~
+
+If the user is simply invoking the parameter sweep tool, they only need to
+specify parallel backend by using the keyword argument ``parallel_back_end``
+when constructing a parameter sweep object. Valid keyword values are
+
+    #. "MPI"
+    #. "ConcurrentFutures"
+    #. "MultiProcessing"
+    #. "RayIo"
+
+If any other keyword is provided, the parallel manager reverts to serial. Note
+that, when using MPI, the end user must still use the ``mpiexec`` or ``mpirun``
+command when executing the parameter sweep from the command line.
+
+Adding Parallel Manager to Your Code
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In order to use the parallel manager in your code, you will need to adapt the
+following statement. 
+
+.. code:: python
+
+    from watertap.tools.parallel.parallel_manager_factory import create_parallel_manager
+    parallel_manager = create_parallel_manager(
+            parallel_manager_class=None,
+            number_of_subprocesses=2,
+            parallel_back_end="ConcurrentFutures",
+        )
+
+The above code return 1 of 5 parallel manager classes. There are currently 5 parallel manager classes:
+
+    #. ``SingleProcessParallelManager`` : To enable serial implementation
+    #. ``MPIParallelManager`` : For MPI
+    #. ``ConcurrentFuturesParallelManager`` : For python concurrent futures
+    #. ``MultiprocessingParallelManager`` : For more advanced python multiprocessing
+    #. ``RayIoParallelManager`` : For using Ray
+
+Alternatively, the user can specify the parameter sweep class directly, for example
+
+.. code:: python
+
+    from watertap.tools.parallel.single_process_parallel_manager import SingleProcessParallelManager
+    s_parallel_manager = create_parallel_manager(
+            parallel_manager_class=SingleProcessParallelManager,
+            number_of_subprocesses=1,
+        )
+
+Each of the parallel manager classes inherits from the base ``ParallelManager``
+class and defines the abstract methods. The methods can be found in the
+class documentation below. 
+
+* :mod:`watertap.tools.parallel.parallel_manager`
+* :mod:`watertap.tools.parallel.parallel_manager_factory`
+* :mod:`watertap.tools.parallel.concurrent_futures_parallel_manager`
+* :mod:`watertap.tools.parallel.mpi_parallel_manager`
+* :mod:`watertap.tools.parallel.multiprocessing_parallel_manager`
+* :mod:`watertap.tools.parallel.ray_io_parallel_manager`
+* :mod:`watertap.tools.parallel.single_process_parallel_manager`
+
+
+Adding Features to the Parallel Manager
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+An advanced user may add features as necessary to the parallel manager. We ask
+that they follow appropriate code development practices and include testing and
+documentation. A lot of features are currently missing from the parallel 
+manager that limits its wider use. We encourage developers to make PRs to the
```

Comment:
Are there some immediate examples that come to mind that we should list somewhere, e.g., in an issue?

Reply:
We have all the necessary features needed to run the parameter sweep. If however, we wanted to make the parallel manager into a standalone package we would need features that, for example, allows one worker to send data to another and so on. I do not think we need to make an issue for it quite yet as this is only tangentially related to WaterTAP.